sympy/integrals/integrals.py

```diff
@@ -417,6 +418,17 @@ def doit(self, **hints):
         if self.is_zero:
             return S.Zero
 
+        # hacks to handle integrals of
+        # nested summations
+        if isinstance(self.function, Sum):
+            if any(v in self.function.limits[0] for v in self.variables):
+                raise ValueError('Limit of the sum cannot be an integration variable.')
+            if any(not l.is_finite for l in self.function.limits[0][1:]):
```

Comment:
I think that `not l.is_finite` should rather be `l.is_infinite`, which is less restrictive. This would change the test with symbol `M`:
```
>>> M = Symbol('M')
>>> not M.is_finite
True
>>> M.is_infinite
>>>
```
It is likely that `M` is used to denote an integer so integration should be allowed. (Otherwise, `oo` or `-oo` would have been used.)